meta-xt-control-domain/recipes-guest/doma/files/doma-restart-monitor-ExecStart.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+echo "DomA restart monitor started!";
+
+# Fetch DomA Xen domain identifier
+DOMA_ID="";
+
+while [ "$DOMA_ID" == "" ]; do
+    DOMA_ID=$(xl list | awk '{ if ($1 == "DomA") print $2 }');
```

Comment:
`xl domid`

Reply:
Hi @lorc, I'm not sure that it will be better.

```
DOMD_ID="";

output=$(xl domid DomD)
if [ $? -eq 0 ]; then
    DOMD_ID=$output
fi
```

Also, how would it work with "set -e"?

Reply:
Hi @lorc, I'll close this point as of now. In case of usage of the "xl domid" command, I'll need to additionally check for the error state (which happens when the domain is not available ) to void assigning the error content to the DOMD_ID variable.

The error looks like this:

```
root@generic-armv8-xt-dom0:~# xl domid non_existing_domain_id
Can't get domid of domain name 'non_existing_domain_id', maybe this domain does not exist.
```

It'll take more lines of code and won't be that elegant.

awk, in its turn, will return an empty string in case of a non-existing domain. That is more suitable in this specific case.